src/tree/tree.rs

```diff
@@ -28,7 +27,6 @@ impl<L, A, const DEPTH: usize> AsRef<NStack<L, A>> for PoseidonTree<L, A, DEPTH>
 where
     L: PoseidonLeaf,
     A: PoseidonTreeAnnotation<L>,
-    A: Annotation<L> + Combine<NStack<L, A>, A>,
```

Comment:
I don't see any reason why removing this is helpful. Maybe I'm missing something but this just seems to remove customization capabilities to the end user as is shown in the tests.

Now you'll not be able to decide how the nodes are combined neither the annotations.

Reply:
As both in the description of this PR and the poseidon annotation documentation, any trait requirement beyond the scope of `PoseidonTreeAnnotation` violates its purpose

https://github.com/dusk-network/Poseidon252/blob/master/src/tree/annotation/mod.rs#L20-L21

No flexibility is lost because if someone wants to customize the behavior of the tree they can just create a new annotation and implement this trait.

Reply:
Yep have to agree, removing unneccesary trait bounds is always a good thing.